client/my-sites/plan-features/index.jsx

```diff
@@ -952,9 +954,16 @@ const ConnectedPlanFeatures = connect(
 		const canPurchase = ! isPaid || isCurrentUserCurrentPlanOwner( state, selectedSiteId );
 		const isLoggedInMonthlyPricing =
 			! isInSignup && ! isJetpack && kindOfPlanTypeSelector === 'interval';
+		const isInHideECommerceExperiment =
+			isInSignup && isLaunchPage && 'treatment' === experimentVariation;
+		const filteredPlans = isInHideECommerceExperiment
```

Comment:
I suggest filtering the plans in the function responsible for it - `getPlansForPlanFeatures()` [here](https://github.com/Automattic/wp-calypso/blob/05732f098d92569f4fb9d8398ade6f9c2006e339/client/my-sites/plans-features-main/index.jsx#L285) so that we can keep related logic in the same place.

Reply:
I had to shuffle around the ProvideExperimentData in order to make this work.

getPlansForPlanFeatures is called twice in the same render(), it's not clear why.